### PR TITLE
Make things build on i386-gnu again

### PR DESCRIPTION
**Pull request by bugaevc**
cc:
* @spholz and @BertalanD for the LibELF change
* @nico for LibPDF & JBIG2
* @ADKaster for `Core::MachPort`
* @stelar7 for LibCrypto / MGF

### CODE REVIEW

**Comment by bugaevc** [comment]
Visual proof that things still work:

Screenshot from 2024-04-24 17-21-05_skip_unrelated to pdf

**Comment by bugaevc** [comment]
> Error: An error occurred trying to start process '/usr/bin/bash' with working directory '/home/runner/work/serenity/serenity/Build/x86_64'. No such file or directory

That's clearly not my fault?

**Comment by bugaevc** [comment]
Pushed a fix for aarch64-gnu.

By the way, please feel free to cherry-pick individual patches, they should be independent from each other.

File: Userland/Libraries/LibCrypto/Hash/MGF.h [hunk 1]
```diff
@@ -23,9 +23,11 @@ class MGF {
 
         size_t h_len = hash.digest_size();
 
+#ifdef __LP64__
```
Comment (ADKaster) [inline Userland/Libraries/LibCrypto/Hash/MGF.h:26]: LP64 isn't quite enough to select "size_t is > 32 bits" . For example, the MSVC ABI is LLP64. Surely we have AK_IS_64_BIT? or similar...

or we could do like to be explicit about the check.

```
if constexpr (sizeof(size_t) > 32)  {
   if (length > (h_len << 32)
       return ...;
}
```

File: Userland/Libraries/LibGfx/ImageFormats/JBIG2Loader.cpp [hunk 2]
```diff
@@ -239,7 +239,7 @@ class BitBuffer {
 
 ErrorOr<NonnullOwnPtr<BitBuffer>> BitBuffer::create(size_t width, size_t height)
 {
-    size_t pitch = ceil_div(width, 8ull);
```
Comment (ADKaster) [inline Userland/Libraries/LibGfx/ImageFormats/JBIG2Loader.cpp:242]: Commit message here leaves a bit to be desired.

"LibGfx: Explicitly cast literals to size_t in JBIG2Loader"
?

File: AK/Platform.h [hunk 3]
```diff
@@ -7,6 +7,10 @@
 
 #pragma once
 
+#if __has_include(<features.h>)
```
Comment (ADKaster) [inline AK/Platform.h:10]: just to confirm, `__has_include` is always a thing on both clang and gcc for the versions we care about (i.e. with c++20 support)? we had to ifdef `__has_feature` or some other thing later in this file.

**Review by ADKaster — COMMENTED** [review]
These changes seem mostly fine. I'd like to ping @DanShaders for the LibELF change, as he's been poking around there lately. Not sure if it'll still be required after all his recent patches or not.

File: AK/Platform.h (hunk 3 above)
Comment (bugaevc) [reply AK/Platform.h:10]: https://en.cppreference.com/w/cpp/preprocessor/include says it's in C++17

File: Userland/Libraries/LibCrypto/Hash/MGF.h (hunk 1 above)
Comment (bugaevc) [reply Userland/Libraries/LibCrypto/Hash/MGF.h:26]: > `if constexpr`

Will that get rid of the compile time error though?

**Comment by DanShaders** [comment]
I think LibELF changes are not required anymore after (merged) #24105. We don't build these files on lagom now.

File: Userland/Libraries/LibCrypto/Hash/MGF.h (hunk 1 above)
Comment (ADKaster) [reply Userland/Libraries/LibCrypto/Hash/MGF.h:26]: yes <https://godbolt.org/z/qc5xqWro1>

Comment (bugaevc) [reply Userland/Libraries/LibCrypto/Hash/MGF.h]: Done

File: Userland/Libraries/LibGfx/ImageFormats/JBIG2Loader.cpp (hunk 2 above)
Comment (bugaevc) [reply Userland/Libraries/LibGfx/ImageFormats/JBIG2Loader.cpp:242]: Changed to that

**Comment by bugaevc** [comment]
> I think LibELF changes are not required anymore after (merged) #24105. We don't build these files on lagom now.

It built fine without the ELF changes indeed, thanks!

**Review by ADKaster — APPROVED** [review]

**Pull request merged**